src/Generating/HeiGen.cpp

```diff
@@ -10,6 +10,66 @@
 #include "DistortedHeightmap.h"
 #include "EndGen.h"
 #include "Noise3DGenerator.h"
+#include "ProtIntGen.h"
+
+
+
+
+
+////////////////////////////////////////////////////////////////////////////////
+// cHeiGenSteppy:
+
+class cHeiGenSteppy:
+	public cTerrainHeightGen
+{
+public:
+	cHeiGenSteppy(int a_Seed) :
+		m_Seed(a_Seed)
+	{
+		m_Gen =
+			std::make_shared<cProtIntGenWeightAvg<16, 1, 0>>(
+			std::make_shared<cProtIntGenSmooth>             (a_Seed + 1,
+			std::make_shared<cProtIntGenZoom>               (a_Seed + 2,
+			std::make_shared<cProtIntGenSmooth>             (a_Seed + 3,
+			std::make_shared<cProtIntGenZoom>               (a_Seed + 4,
+			std::make_shared<cProtIntGenAddRnd>             (a_Seed + 5, 1,
+			std::make_shared<cProtIntGenSmooth>             (a_Seed + 6,
+			std::make_shared<cProtIntGenZoom>               (a_Seed + 7,
+			std::make_shared<cProtIntGenRndBetween>         (a_Seed + 8, 60,
+			std::make_shared<cProtIntGenAddRnd>             (a_Seed + 9, 1,
+			std::make_shared<cProtIntGenSmooth>             (a_Seed + 1,
+			std::make_shared<cProtIntGenZoom>               (a_Seed + 2,
+			std::make_shared<cProtIntGenRndBetween>         (a_Seed + 3, 60,
+			std::make_shared<cProtIntGenSmooth>             (a_Seed + 4,
+			std::make_shared<cProtIntGenZoom>               (a_Seed + 5,
+			std::make_shared<cProtIntGenRndBetween>         (a_Seed + 6, 60,
+			std::make_shared<cProtIntGenRndChoice>          (a_Seed + 7, 10, 50, 50,
+			std::make_shared<cProtIntGenSmooth>             (a_Seed + 8,
+			std::make_shared<cProtIntGenZoom>               (a_Seed + 9,
+			std::make_shared<cProtIntGenRndChoice>          (a_Seed + 1, 10, 50, 50,
+			std::make_shared<cProtIntGenAddRnd>             (a_Seed + 2, 2,
+			std::make_shared<cProtIntGenZoom>               (a_Seed + 3,
+			std::make_shared<cProtIntGenZoom>               (a_Seed + 4,
+			std::make_shared<cProtIntGenChoice>             (a_Seed + 5, 10)
+		)))))))))))))))))))))));
```

Comment:
This has reminded me that I'm still not happy about this pipeline syntax, but that's a separate issue.